python/pyspark/sql/tests.py

```diff
@@ -429,6 +429,11 @@ def test_udf_with_input_file_name(self):
         row = self.spark.read.json(filePath).select(sourceFile(input_file_name())).first()
         self.assertTrue(row[0].find("people1.json") != -1)
 
+    def test_udf_should_validate_input_args(self):
+        from pyspark.sql.functions import udf
+
+        self.assertRaises(TypeError, udf(lambda x: x), None)
```

Comment:
Yeah. I am afraid it can actually cause more troubles than its worth:
- If we throw an exception there is a chance we hit some border cases.
- Issuing a warning doesn't prevent task failure so it doesn't provide the same advantages as failing early.

Maybe it is better to leave it as is. Right now users get a clear feedback, if there is an incorrect type, and for additional safety one can always use annotations and type checker.